### PR TITLE
XWIKI-19826: Move the old event store to legacy

### PR DESCRIPTION
**Pull request by tmortagne**
There are two aspects:
* make sure we don't rely on the old events store or its APIs in XWiki Standard
* move everything which is related to the old event store and its APIs in a single extension that we could later stop bundling and let user install it if needed instead

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/script/EventScriptService.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.xwiki.eventstream.EventStreamException;
+import org.xwiki.eventstream.RecordableEventDescriptor;
+import org.xwiki.eventstream.RecordableEventDescriptorManager;
+import org.xwiki.script.service.ScriptService;
+
+/**
+ * Script services for the Event Stream Module.
+ *
+ * @version $Id$
+ * @since 9.2RC1
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/script/EventScriptService.java:38]: I'm probably missing something. How can a new script service have this "since"?

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/script/EventScriptService.java:38]: > How can a new script service have this "since"?

With missing fix after a copy/paste :)

Fixing that.

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministration.xml [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+{{info}}
+  {{translation key="legacy.event.count.store"/}}: **$services.eventstream.eventCount**
+  {{translation key="legacy.event.count.legacy"/}}: **$services.eventstream.legacyEventCount**
+{{/info}}
+
+{{html}}
+&lt;form id="migrate" action="$escapetool.xml($xwiki.relativeRequestURL)" method="post" class="xform third"&gt;
+  &lt;div&gt;
+    &lt;input type="radio" id="mode_alltime" name="mode" value="alltime"#if($request.mode != 'since')checked#end/&gt;
+    &lt;label for="mode_alltime"&gt;$services.localization.render('legacy.event.migration.alltime')&lt;/label&gt;
```
[the remark this answers is not in the document]
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministration.xml:65]: Honestly, I would prefer to keep this pull request focused on what it's about, don't hesitate to create a jira issue for this.

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml [hunk 3]
```diff
@@ -24,22 +24,21 @@
...
     <parent>
         <groupId>org.xwiki.platform</groupId>
-        <artifactId>xwiki-platform-eventstream-stores</artifactId>
+        <artifactId>xwiki-platform-legacy-events-hibernate</artifactId>
         <version>14.5-SNAPSHOT</version>
     </parent>
-    <artifactId>xwiki-platform-eventstream-store-hibernate</artifactId>
-    <name>XWiki Platform - Event Stream - Store - Hibernate</name>
-    <description>Hibernate based events storage</description>
+    <artifactId>xwiki-platform-legacy-events-hibernate-api</artifactId>
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml:30]: Why this name choice? why not keeping `eventstream-store-hibernate` and why adding api there?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml [hunk 4]
```diff
@@ -24,22 +24,21 @@
...
     <properties>
-        <xwiki.jacoco.instructionRatio>0.18</xwiki.jacoco.instructionRatio>
-        <!-- Name to display by the Extension Manager -->
-        <xwiki.extension.name>Hibernate Event Stream Store</xwiki.extension.name>
+        <xwiki.jacoco.instructionRatio>0.20</xwiki.jacoco.instructionRatio>
 
         <xwiki.extension.features>
           <!-- Previous name of the module -->
           org.xwiki.platform:xwiki-platform-eventstream-store
+          org.xwiki.platform:xwiki-platform-legacy-eventstream-store-hibernate
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml:39]: you meant `xwiki-platform-eventstream-store-hibernate` here no?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/notifications/sources/internal/LegacyEventSearcher.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+
+/**
+ * Search {@link Event}.
+ * 
+ * @version $Id$
+ * @since 14.6RC1
+ */
+@Component(roles = EventSearcher.class)
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/notifications/sources/internal/LegacyEventSearcher.java:43]: It means it overrides the default EventSearcher component in the CM? There couldn't be any problem of which one is loaded first?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministrationTranslations.xml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="XWiki.Notifications.Code.LegacyNotificationAdministrationTranslations" locale="">
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministrationTranslations.xml:23]: Don't forget to create the component on weblate once merged.

File: xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/pom.xml [hunk 7]
```diff
@@ -54,12 +54,6 @@
       <artifactId>xwiki-platform-notifications-notifiers-api</artifactId>
       <version>${project.version}</version>
     </dependency>
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/pom.xml:56]: I think you forgot to skip revapi in this module since now it should be executed in the legacy module.

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml [hunk 8]
```diff
@@ -31,7 +31,7 @@
   <name>XWiki Platform - Notifications - Sources</name>
   <description>Handle how notifications are retrieved</description>
   <properties>
-    <xwiki.jacoco.instructionRatio>0.78</xwiki.jacoco.instructionRatio>
+    <xwiki.jacoco.instructionRatio>0.71</xwiki.jacoco.instructionRatio>
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml:34]: Why did you remove `DefaultParametrizedNotificationManagerTest` in this module?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java [hunk 9]
```diff
@@ -56,6 +56,7 @@
 @Component
 @Singleton
 @Named("eventstream")
+@Deprecated
 public class EventStreamScriptService implements ScriptService
```
Comment (surli) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java:60]: it cannot extends the new `EventScriptService` since you implemented there some APIs of it?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml (hunk 4 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml:39]: Yes, fixing.

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/notifications/sources/internal/LegacyEventSearcher.java (hunk 5 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/notifications/sources/internal/LegacyEventSearcher.java:43]: There are two possibilities:
* it's a core extension (the current situation): there is an override in the component.txt
* it's an installed extension: it's always registered after the other one, so the legacy one wins

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministrationTranslations.xml (hunk 6 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-ui/src/main/resources/XWiki/Notifications/Code/LegacyNotificationAdministrationTranslations.xml:23]: Yes, it's planned.

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml (hunk 8 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml:34]: Because despite its very generic name, all the tests were actually relying on the old store. Did not had time to rewrite it entirely, so I just moved it.

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java (hunk 9 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java:60]: It's not very clear what you mean. Are you talking about keeping "eventstream" for EventScriptService ?

Comment (surli) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java:60]: No I meant to write `public class EventStreamScriptService extends EventScriptService`, it looks like the code of `getRecordableEventDescriptors` and `getDescriptorForEventType` is now duplicated no?

File: xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml (hunk 8 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-sources/pom.xml:34]: > so I just moved it.

you mean removed it? Or did I miss the new test?

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml (hunk 3 above)
Comment (surli) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/pom.xml:30]: ok

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java (hunk 9 above)
Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/main/java/org/xwiki/eventstream/script/EventStreamScriptService.java]: This made me think of a less disturbing change: I reverted the changed on standard side and just moved the legacy related method to override of the script service on legacy side. That way the script service ID does not change.

**Comment by surli** [comment]
Just a question: do you plan to provide a snippet for the migration of events? I guess it could still be useful in some cases to be able to implement the migration if it's not available anymore from the admin.

**Comment by tmortagne** [comment]
IMO if you need to migrate event you would install this extension which also put a migration UI in the admin.

**Comment by surli** [comment]
> IMO, if you need to migrate events after we removed this legacy extension (which is not planned before at least 15.x) then you should install it, and you will get a migration UI in the admin.

Ok I wrongly thought you removed the `LegacyEventMigrationJob` but it's actually the `LegacyEventMigrationJobTest` (here: https://github.com/xwiki/xwiki-platform/pull/1856/files#diff-34b908fb19c903d2d4a94f996b4152a639797ee453fbe839b5cbd502897f9c33) it's expected to remove it?

**Comment by tmortagne** [comment]
> it's expected to remove it?

No, looks like another case of test class lost when I merged all the legacy modules in one. Checking.

File: xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/test/java/org/xwiki/eventstream/internal/LegacyEventMigrationJobTest.java [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link LegacyEventMigrationJob}.
+ *
+ * @version $Id$
+ */
+@ComponentTest
+public class LegacyEventMigrationJobTest
```
Comment (manuelleduc) [inline xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/test/java/org/xwiki/eventstream/internal/LegacyEventMigrationJobTest.java:62]: useless public keyword ;)

Comment (tmortagne) [reply xwiki-platform-core/xwiki-platform-legacy/xwiki-platform-legacy-events-hibernate/xwiki-platform-legacy-events-hibernate-api/src/test/java/org/xwiki/eventstream/internal/LegacyEventMigrationJobTest.java:62]: I thought about fixing it but since it was just a file move...

File: pom.xml [hunk 11]
```diff
@@ -58,7 +58,7 @@
     <pdfbox.version>2.0.26</pdfbox.version>
     <keypress.version>2.1.5</keypress.version>
     <jetty.version>9.4.29.v20200521</jetty.version>
-    <netty.version>4.1.77.Final</netty.version>
+    <netty.version>4.1.78.Final</netty.version>
```
Comment (manuelleduc) [inline pom.xml:61]: Is this version upgrade part of the same issue?

Comment (tmortagne) [reply pom.xml:61]: No, let me merge master branch to not have it listed anymore.
